### DISCUSSION TITLE
Does pyright detect the conditional dependencies when linting?

### DISCUSSION THREAD

**Discussion by kemingy** [microsoft/pyright#8926] [Q&A] [answered]
I need to use [`importlib.resources.files`](https://docs.python.org/3/library/importlib.resources.html#importlib.resources.files) in my project, which is added in Python 3.9. For compatibility, I added `"importlib_resources >= 5.10; python_version < '3.12'"` to my pyproject file.

But pyright keep reporting the `files` cannot be found in the `importlib.resources` since the `pythonVersion = "3.8"`.

I'd like to know if pyright can detect the conditional dependencies.

**Comment by erictraut** [comment, marked as answer]
The typeshed stub file for the `importlib.resources` module indicates that `files` is available for Python 3.9 and newer. See [this link](https://github.com/python/typeshed/blob/61ed105cdbc2586034039a617d66fc1453630ee5/stdlib/importlib/resources/__init__.pyi#L36) for details.

If pyright indicates that it can't find the `files` symbol, that's probably because you've configured it to assume Python 3.8 or older. By default, python will use the Python version from the python environment you've configured from the command-line shell. Type `python3 --version` to see the version of the active python environment. If it's not what you're expecting, then activate the intended python environment before running pyright.
